lib/sandboxed_renderer/init.js

```diff
@@ -40,14 +52,18 @@ preloadProcess.hang = () => binding.hang()
 preloadProcess.getProcessMemoryInfo = () => binding.getProcessMemoryInfo()
 preloadProcess.getSystemMemoryInfo = () => binding.getSystemMemoryInfo()
 preloadProcess.argv = binding.getArgv()
-process.platform = preloadProcess.platform = electron.remote.process.platform
-process.execPath = preloadProcess.execPath = electron.remote.process.execPath
+preloadProcess.platform = process.platform = platform
+preloadProcess.execPath = process.execPath = execPath
+
 process.on('exit', () => preloadProcess.emit('exit'))
 
 // This is the `require` function that will be visible to the preload script
 function preloadRequire (module) {
-  if (preloadModules.has(module)) {
-    return preloadModules.get(module)
+  if (loadedModules.has(module)) {
+    return loadedModules.get(module)
+  }
+  if (remoteModules.has(module)) {
+    return require(module)
```

Comment:
I don't have a strong opinion on whether to cache these. My uninformed guess is that the Map lookup would be cheaper than multiple `require()` calls but I don't have numbers to back it up :smiley:  It's also an open question how often modules get re-required.

However, I'm still interested in understanding this patch better -- what's the criteria for putting a module in `loadedModules` vs `remoteModules`? That is, how to decide which modules to preload and which to lazy-load?

Reply:
`remoteModules` are modules which are proxies to node.js modules in the browser process. We do this because the browserify-provided shims for some modules
(child_process, os, fs and path) are not ideal for Electron apps.

Here's how browserify is invoked by Electron build system: https://github.com/electron/electron/blob/master/electron.gyp#L513-L519

These arguments provide replacements for browserify shims, but they also cause electron's `remote` module to be loaded, which is what we want to avoid in early startup for sandbox 

`loadedModules` are browserify shims that are already available in the renderer process, so it is possible for an Electron app to create a preload bundle and pass `-e` to exclude from their own bundle, and the `require` will work since it is provided by `loadedModules`.

Reply:
these things should have been discussed in the original PR. this is a backport.